spras/containers.py

```diff
@@ -148,6 +150,42 @@ def run_container(framework: str, container_suffix: str, command: List[str], vol
     else:
         raise ValueError(f'{framework} is not a recognized container framework. Choose "docker" or "singularity".')
 
+def run_container_and_log(name: str, framework: str, container_suffix: str, command: List[str], volumes: List[Tuple[PurePath, PurePath]], working_dir: str, environment: str = 'SPRAS=True'):
+    """
+    Runs a command in the container using Singularity or Docker with associated pretty printed messages.
+    @param framework: singularity or docker
+    @param container_suffix: name of the DockerHub container without the 'docker://' prefix
+    @param command: command to run in the container
+    @param volumes: a list of volumes to mount where each item is a (source, destination) tuple
+    @param working_dir: the working directory in the container
+    @param environment: environment variables to set in the container
+    @return: output from Singularity execute or Docker run
+    """
+    print('Running {} on framework "{}" with command: {}'.format(name, framework, ' '.join(command)), flush=True)
+    try:
+        out = run_container(framework=framework, container_suffix=container_suffix, command=command, volumes=volumes, working_dir=working_dir, environment=environment)
+        if out is not None:
+            if isinstance(out, list):
+                out = ''.join(out)
+            elif isinstance(out, dict):
+                if 'message' in out:
+                    # singularity message - lets print it.
```

Comment:
It'd be useful drop a comment here that points to documentation describing whatever apptainer output format you're tapping into.